packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -336,6 +336,52 @@ export const getCartDetails = payload => {
     };
 };
 
+export const retrieveAndMergeCarts = payload =>
+    async function thunk(dispatch) {
+        const { mergeCarts, fetchCartId } = payload;
+        const sourceCartId = await retrieveCartId();
+        try {
+            await dispatch(removeCart());
```

Comment:
I see here that we have to remove/create to get the user's cart id as the destination. It wasn't clear at first, so please add some comments here explaining this.

Reply:
Removed usage of redux.